test/dlc_tests/sanity/test_pip_check.py

```diff
@@ -1,18 +1,27 @@
+import re
+
 import pytest
 
-from test.test_utils import run_subprocess_cmd
+from invoke.context import Context
 
 
 @pytest.mark.canary("Run pip check test regularly on production images")
 def test_pip_check(image):
     """
     Test to run pip sanity tests
     """
-    if "tensorflow-inference" in image:
-        pytest.xfail(
-            reason="Tensorflow serving api requires tensorflow, but we explicitly do not install"
-            "tensorflow in serving containers."
-        )
-
     # Add null entrypoint to ensure command exits immediately
-    run_subprocess_cmd(f"docker run --entrypoint='' {image} pip check")
+    ctx = Context()
+    gpu_suffix = '-gpu' if 'gpu' in image else ''
+
+    # TF inference containers do not have core tensorflow installed by design. Allowing for this pip check error
+    # to occur in order to catch other pip check issues that may be associated with TF inference
+    allowed_exception = re.compile(rf'^tensorflow-serving-api{gpu_suffix} \d\.\d+\.\d+ requires '
+                                   rf'tensorflow{gpu_suffix}, which is not installed.$')
+    output = ctx.run(f"docker run --entrypoint='' {image} pip check", hide=True, warn=True)
```

Comment:
Should we use "nvidia-docker" for gpu images, or it should be no difference since it is just a pip check?

Reply:
it shouldn't matter for pip check since we are not using any gpu specific functionality